### PR TITLE
Doc updates 24q1 2

### LINKED ISSUE #1080: Minor doc/tutorial additions.
Just noticed we did not entirely update the docs/tutorials for some additions at the last release (`Coef` and SDP for example).

https://computationalcryoem.github.io/ASPIRE-Python/auto_tutorials/aspire_introduction.html#pipeline-roadmap

I'll try to skim through docs and make a few touch ups.

### PR DESCRIPTION
**Pull request by garrettwrong**
Closes #1080 

Minor update to description of class averaging in demo pipeline.
Resolves ImageSource docstrings that failed to render due to use of decorator.
Removes superfluous ArrayImageSource in example.

### CODE REVIEW

File: gallery/tutorials/pipeline_demo.py [hunk 1]
```diff
@@ -250,15 +252,13 @@
...
-# ``ArrayImageSource`` to access the ``Image.show()`` method.
-projections_est = ArrayImageSource(estimated_volume.project(oriented_src.rotations))
+# Get the first 10 projections from the estimated volume using the
+# estimated orientations.  Recall that ``project`` returns an
+# ``Image`` instance, which we can peek at with ``show``.
+projections_est = estimated_volume.project(oriented_src.rotations[0:10])
 
 # We view the first 10 projections of the estimated volume.
-projections_est.images[0:10].show()
+projections_est.show()
```
Comment (j-c-c) [inline gallery/tutorials/pipeline_demo.py:262]: Thanks!

File: src/aspire/source/image.py [hunk 2]
```diff
@@ -114,6 +115,7 @@ def as_copy(func):
     is best used on methods that mutate the object but don't return anything.
     """
 
+    @functools.wraps(func)  # Pass metadata (eg name and doctrings) from `func`
```
Comment (j-c-c) [inline src/aspire/source/image.py:118]: This is great. Docs look good.

Should we make `as_copy` private? Currently it appears in the docs.

**Review by j-c-c — COMMENTED** [review]
Looks good! Just one question.

File: src/aspire/source/image.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/source/image.py:118]: lol, I'd rather delete it, but sure, I'll make a commit with that.

**Review by j-c-c — APPROVED** [review]
LGTM!

**Comment by garrettwrong** [comment]
These are pretty minor, I think we can merge ourselves. Merging.

**Pull request merged**